### PR TITLE
PillButtonBar: Fixing warnings for iOS 15 target deployment version.

### PR DESCRIPTION
**Pull request by rdeassis**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

We're soon going to be bumping our target deployment version to iOS 15.
This will trigger compile time warnings regarding deprecated APIs in iOS 15.

Because some of our client apps have the compiler configured to treat warnings as errors, we have to fix these warnings in advance so they won't get build breaks when the target deployment version is updated.

This PR addresses the PillButtonBar that now needs to use the UIButton Configuration to set title, background color and content insets.

### Verification

Exercised scenarios in the demo app to ensure no regressions were caused:

https://user-images.githubusercontent.com/68076145/184774897-021dea42-b3ef-410c-8f3e-5d38e069d7ff.mov

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [x] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/1165)

### CODE REVIEW

File: ios/FluentUI/Pill Button Bar/PillButtonBar.swift [hunk 1]
```diff
@@ -125,6 +125,7 @@ open class PillButtonBar: UIScrollView {
 
     private var stackView: UIStackView = {
         let view = UIStackView()
+        view.distribution = .fillProportionally
```
Comment (harrieshin) [inline ios/FluentUI/Pill Button Bar/PillButtonBar.swift:128]: why this change?

Comment (rdeassis) [reply ios/FluentUI/Pill Button Bar/PillButtonBar.swift:128]: The buttons using the configuration were now the same width by default. The stack view gets the widest button and apply that same width for all the buttons unless the distribution .fillProportionally is set.

Comment (harrieshin) [reply ios/FluentUI/Pill Button Bar/PillButtonBar.swift:128]: i see. then shouldn't it be only set for ios15 and newer?

Comment (rdeassis) [reply ios/FluentUI/Pill Button Bar/PillButtonBar.swift:128]: I can make it specific for that, but it seems like it'll be just redundant in the worst case scenario for iOS 14. It didn't cause any regressions for iOS 14.
Let me know your preference. I'm fine with either option.

Comment (harrieshin) [reply ios/FluentUI/Pill Button Bar/PillButtonBar.swift:128]: i see as long as it didn't cause regression in iOS 14 and the scrolling calculation

**Review by harrieshin — APPROVED** [review]

**Comment by rdeassis** [comment]
> test RTL and larger text?

Yes!

Large text:
Screen Shot 2022-08-17 at 9 02 45 AM

RTL:
Screen Shot 2022-08-17 at 9 03 59 AM

File: ios/FluentUI/Pill Button Bar/PillButtonBar.swift (hunk 1 above)
Comment (rdeassis) [reply ios/FluentUI/Pill Button Bar/PillButtonBar.swift:128]: Yes. I validated on both iOS 14 and 15 and no regressions were introduced by this extra call.
